katello/katello/hostname-change.rb

```diff
@@ -363,25 +369,63 @@ def run
         end
         installer << " --certs-regenerate=true --foreman-proxy-register-in-foreman true"
       end
-      # always disable system checks to avoid unnecessary errors. The installer should have
-      # already ran since this is to be run on an existing system and installer checks would
-      # have already been skipped
-      installer << " --disable-system-checks" if disable_system_check_option?
 
       STDOUT.puts installer
-      installer_output = self.run_cmd("#{installer}")
-      installer_success = $?.success?
-      STDOUT.puts installer_output
-
-      STDOUT.puts "Restarting puppet services"
-      self.run_cmd("/sbin/service puppet restart")
-      self.run_cmd("katello-service restart --only puppetserver")
+      run_cmd(installer, [0], installer_failure_message) do |result, success|
+        if temp_last_scenario_yaml && temp_last_scenario_yaml.path
```

Comment:
Won't this throw a ```NameError``` in the case where the file didn't exist? temp_last_scenario_yaml is only assigned inside the if block.

Reply:
`temp_last_scenario_yaml` will be nil and the `temp_last_scenario_yaml.path` will not be executed if I am understanding your question so it should be good in that scenario

Reply:
I'm talking about:
https://github.com/jturel/foreman-packaging/blob/741cdd39f53af493dc9e99cf88ec9071ceb1f8bc/katello/katello/hostname-change.rb#L344-L355

If false, then the variable isn't nil but actually unset I think.

Reply:
Ah, yes. I'll adjust that!